include/s3select_oper.h

```diff
@@ -1464,32 +1472,31 @@ class logical_operand : public base_statement
     value a = l->eval();
     if (_oplog == oplog_t::AND)
     {
-      if (a.i64() == false)
-      {
+      if (a.i64() == false) {
         res = false ^ negation_result;
         return var_value = res;
       } 
       value b = r->eval();
+      res =  (a.i64() && b.i64()) ^ negation_result ;
+      
       if (a.is_null() || b.is_null()) {
         var_value.setnull();
-      } else {
-        res =  (a.i64() && b.i64()) ^ negation_result ;
-      }
+        return var_value;
+      } 
     }
     else
     {
-      if (a.i64() == true)
-      {
+      if (a.i64() == true) {
```

Comment:
flow is not correct , the following wont result correctly 
/where (1 <= null) or (1==1);/